lib/Sema/DerivedConformanceAdditiveArithmeticVectorNumeric.cpp

```diff
@@ -154,16 +154,16 @@ bool DerivedConformance::canDeriveAdditiveArithmetic(NominalTypeDecl *nominal,
   // All stored properties must conform to `AdditiveArithmetic`.
   auto &C = nominal->getASTContext();
   auto *addArithProto = C.getProtocol(KnownProtocolKind::AdditiveArithmetic);
-    return llvm::all_of(structDecl->getStoredProperties(), [&](VarDecl *v) {
-      if (!v->hasInterfaceType() || !v->getType())
-        C.getLazyResolver()->resolveDeclSignature(v);
-      if (!v->hasInterfaceType() || !v->getType())
-        return false;
-      auto declType = v->getType()->hasArchetype()
-          ? v->getType()
-          : DC->mapTypeIntoContext(v->getType());
-      return (bool)TypeChecker::conformsToProtocol(declType, addArithProto, DC,
-                                                   ConformanceCheckFlags::Used);
+  return llvm::all_of(structDecl->getStoredProperties(), [&](VarDecl *v) {
+    if (!v->getType())
+      C.getLazyResolver()->resolveDeclSignature(v);
+    if (!v->getType())
+      return false;
+    auto declType = v->getType()->hasArchetype()
+        ? v->getType()
+        : DC->mapTypeIntoContext(v->getType());
```

Comment:
mapTypeIntoContext() is a no-op when hasArchetype() is false

Reply:
Thank you! Simplified in https://github.com/apple/swift/pull/22249/commits/e5d89ca2225027d68050e426ee1dc79b75bf881a.